### ISSUE TITLE
False positive for unreachable_code_linter() on inline expressions

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2259]
```r
lint("function(x) if (x > 2) stop() else x\n", unreachable_code_linter())
# <text>:1:31: warning: [unreachable_code_linter] Code and comments coming after a return() or stop() should be removed.
# function(x) if (x > 2) stop() else x
#                               ^~~~
```

This may be a regression; cc @MEO265.

**Comment by MEO265** [comment]
Maybe we can use the following in line 66 of the linter, then it would no longer be marked and shouldn't insert any false positives, I think
`(//FUNCTION | //OP-LAMBDA)[following-sibling::expr[1]/*[1][self::OP-LEFT-BRACE]]/following-sibling::expr[1]`
I've already written tests and it works
